simpletransformers/seq2seq/seq2seq_model.py

```diff
@@ -0,0 +1,831 @@
+            **kwargs (optional): For providing proxies, force_download, resume_download, cache_dir and other options specific to the 'from_pretrained' implementation where this will be supplied.
+        """  # noqa: ignore flake8"
+
+        if not config:
+            # if not ((encoder_name and decoder_name) or encoder_decoder_name) and not encoder_type:
+            if not ((encoder_name and decoder_name) or encoder_decoder_name):
+                raise ValueError(
+                    "You must specify a Seq2Seq config \t OR \t"
+                    "encoder_type, encoder_name, and decoder_name OR \t \t"
+                    "encoder_type and encoder_decoder_name"
+                )
+            elif not encoder_type:
+                raise ValueError(
+                    "You must specify a Seq2Seq config \t OR \t"
+                    "encoder_type, encoder_name, and decoder_name \t OR \t"
+                    "encoder_type and encoder_decoder_name"
+                )
+
+        if args and "manual_seed" in args:
+            random.seed(args["manual_seed"])
+            np.random.seed(args["manual_seed"])
+            torch.manual_seed(args["manual_seed"])
+            if "n_gpu" in args and args["n_gpu"] > 0:
+                torch.cuda.manual_seed_all(args["manual_seed"])
+
+        self.args = {
+            "dataset_class": None,
+            "do_sample": False,
+            "max_steps": -1,
+            "evaluate_generated_text": False,
+            "num_beams": 1,
+            "max_length": 20,
+            "repetition_penalty": 1.0,
+            "length_penalty": 2.0,
+            "early_stopping": True,
+            "preprocess_inputs": True,
+        }
+
+        self.args.update(global_args)
+
+        if args:
+            self.args.update(args)
+
+        if use_cuda:
+            if torch.cuda.is_available():
+                if cuda_device == -1:
+                    self.device = torch.device("cuda")
+                else:
+                    self.device = torch.device(f"cuda:{cuda_device}")
+            else:
+                raise ValueError(
+                    "'use_cuda' set to True when cuda is unavailable."
+                    "Make sure CUDA is available or set `use_cuda=False`."
+                )
+        else:
+            self.device = "cpu"
+
+        self.results = {}
+
+        if not use_cuda:
+            self.args["fp16"] = False
+
+        # config = EncoderDecoderConfig.from_encoder_decoder_configs(config, config)
+        config_class, model_class, tokenizer_class = MODEL_CLASSES[encoder_type]
+
+        if encoder_decoder_name:
+            # self.model = EncoderDecoderModel.from_pretrained(encoder_decoder_name)
+            self.model = EncoderDecoderModel.from_encoder_decoder_pretrained(
+                os.path.join(encoder_decoder_name, "encoder"), os.path.join(encoder_decoder_name, "decoder")
+            )
```

Comment:
```suggestion
            self.model = EncoderDecoderModel.from_pretrained(encoder_decoder_name)
            # self.model = EncoderDecoderModel.from_encoder_decoder_pretrained(
            #     os.path.join(encoder_decoder_name, "encoder"), os.path.join(encoder_decoder_name, "decoder")
            # )
```
Making this change changes the weights in the loaded model. I have no idea how this can happen. In fact, any combination of the 3 ways to load the model (`from_pretrained`, `from_encoder_decoder_pretrained`, loading encoder/decoder separately) loads different weights to the model!

Reply:
> this is a strange error, because it only occurs after the model has been saved.
> before saving, the predicting worked fine and now returns different results than before saving
> but thats a generally problem, I think, nothing you could change

This is why I wasn't sure the model saving/loading is working correctly.

Reply:
I think it's an issue for the huggingface repo
using `model1 = Seq2SeqModel("outputs/encoder", "outputs/decoder", args=model_args, use_cuda = True,)` I get better results than before saving

Reply:
I think so too. I guess we'll have to see if Hugginface changes anything to deal with this.